internal/comm/catchup_client_test.go

```diff
@@ -0,0 +1,395 @@
+// Copyright IBM Corp. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package comm_test
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"os"
+	"path"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/IBM-Blockchain/bcdb-server/config"
+	"github.com/IBM-Blockchain/bcdb-server/internal/comm"
+	"github.com/IBM-Blockchain/bcdb-server/internal/comm/mocks"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/logger"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewCatchUpClient(t *testing.T) {
+	lg, err := logger.New(&logger.Config{
+		Level:         "debug",
+		OutputPath:    []string{"stdout"},
+		ErrOutputPath: []string{"stderr"},
+		Encoding:      "console",
+	})
+	require.NoError(t, err)
+
+	h := comm.NewCatchUpClient(lg)
+	require.NotNil(t, h)
+}
+
+func TestCatchUpClient_UpdateMembers(t *testing.T) {
+	lg, err := logger.New(&logger.Config{
+		Level:         "debug",
+		OutputPath:    []string{"stdout"},
+		ErrOutputPath: []string{"stderr"},
+		Encoding:      "console",
+	})
+	require.NoError(t, err)
+
+	h := comm.NewCatchUpClient(lg)
+	require.NotNil(t, h)
+
+	peer1 := &types.PeerConfig{
+		NodeId:   "node1",
+		RaftId:   1,
+		PeerHost: "127.0.0.1",
+		PeerPort: 9001,
+	}
+
+	peer2 := &types.PeerConfig{
+		NodeId:   "node2",
+		RaftId:   2,
+		PeerHost: "127.0.0.1",
+		PeerPort: 9002,
+	}
+	err = h.UpdateMembers([]*types.PeerConfig{peer1, peer2})
+	require.NoError(t, err)
+
+	peer2.PeerHost = "not a legal address"
+	err = h.UpdateMembers([]*types.PeerConfig{peer1, peer2})
+	require.EqualError(t, err, "failed to convert PeerConfig [node_id:\"node2\" raft_id:2 peer_host:\"not a legal address\" peer_port:9002 ] to url: parse \"http://not a legal address:9002\": invalid character \" \" in host name")
+}
+
+func TestCatchUpClient_GetHeight(t *testing.T) {
+	lg, err := logger.New(&logger.Config{
+		Level:         "debug",
+		OutputPath:    []string{"stdout"},
+		ErrOutputPath: []string{"stderr"},
+		Encoding:      "console",
+	})
+	require.NoError(t, err)
+
+	localConfigs, sharedConfig := newTestSetup(2)
+
+	ledger1 := &memLedger{}
+	height := uint64(5)
+	for n := uint64(1); n <= height; n++ {
+		ledger1.Append(&types.Block{Header: &types.BlockHeader{BaseHeader: &types.BlockHeaderBase{Number: n}}})
+	}
+	cl1 := &mocks.ConsensusListener{}
+	tr1 := comm.NewHTTPTransport(&comm.Config{
+		LocalConf:    localConfigs[0],
+		Logger:       lg,
+		LedgerReader: ledger1,
+	})
+	require.NotNil(t, tr1)
+	err = tr1.SetConsensusListener(cl1)
+	require.NoError(t, err)
+	err = tr1.UpdateClusterConfig(sharedConfig)
+	require.NoError(t, err)
+
+	err = tr1.Start()
+	require.NoError(t, err)
```

Comment:
Can we modify `startTransportWithLedger` func to by used here as well?

Reply:
done